src/reporters/udp_sender.js

```diff
@@ -99,18 +105,19 @@ export default class UDPSender {
             return { err: true, numSpans: 1 };
         }
 
-        this._byteBufferSize += spanSize;
-        if (this._byteBufferSize <= this._maxSpanBytes) {
+        if (this._totalSpanBytes + spanSize <= this._maxSpanBytes) {
             this._batch.spans.push(span);
-            if (this._byteBufferSize < this._maxSpanBytes) {
+            this._totalSpanBytes += spanSize;
+            if (this._totalSpanBytes < this._maxSpanBytes) {
```

Comment:
why is this extra logic is needed? ie if totalspanbytes == maxspanbytes, the next time round it will be flushed on L118. Do we really need this extra check that has only a 1 byte tolerance?

Reply:
I think you're right. We can just check for strict `<` in L108 and return without flushing

Reply:
* if `total + spanSize` is less than max packet size, we don't flush (diff could be 1 byte or 50Kb, whichever)
  * if `total + spanSize` is exactly the max packet size, we can flush the buffer _including_ the current span, rather than flushing a smaller buffer and keeping the current span for the next flush

minor optimization, but seems worth it given that it doesn't complicate the code that much.